src/snowflake/snowpark/session.py

```diff
@@ -2749,4 +2749,35 @@ def _explain_query(self, query: str) -> Optional[str]:
             _logger.warning("query `%s` cannot be explained", query)
             return None
 
+    def create(self) -> "Session":
+        """Gets the existing session if one exists (in case of SPs/Streamlits) or creates a new one using default connection params (e.g., in case of SPCS)."""
+
+        # 1) check if in SiS or sproc
+        if (
+            is_in_stored_procedure()
+        ):  # (will be removed before merging) no need for `"streamlit" in sys.modules` check since is_in_stored_procedure() must be true for SiS as well
+            # Basically noop and return the existing session
+            return _get_active_session()
+        else:
+            try:
+                # 2) check if we are running in SPCS
+                with open("/snowflake/session/token") as file:
```

Comment:
We could replace this block by adding a helper method like `_get_default_SPSC_connection_params` in connector (similar to `_get_default_connection_params`) to get SPCS default params and move the whole thing in `_create_internal`. Please lemme know which one is a better option/pattern.

Reply:
I have no preference, either will work!